var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -178,7 +176,7 @@ def setup_dependent_build_environment(self, env, dependent_spec):
             env.prepend_path("PATH", self._nc_config_backup_dir)
 
 
-class BackupStep:
+class BackupStep(metaclass=spack.builder.PhaseCallbacksMeta):
```

Comment:
I would prefer `BackupStep` and `Setup` classes to be merged into one. Possibly, under a generic name like `BaseBuilder`. But I can do that later.

Reply:
That's fine with me. The caveat here is that if a mixin has some `@run_after` or `@run_before` decorated method, then it must inherit from the correct metaclass.